cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -113,7 +115,8 @@ public CaReconciler(
             ResourceOperatorSupplier supplier,
             Vertx vertx,
             CertManager certManager,
-            PasswordGenerator passwordGenerator
+            PasswordGenerator passwordGenerator,
+            Clock clock
```

Comment:
> It seems to be very noisy - especially in tests which in 99% of cases don't care about the time at all. 

That's a good point. It's an oversight, I'll change this to be injected with a setter and stay out of the constructor signature.

Reply:
But why do you need a setter? You are just replacing Date with Clock. I would assume that all places which need to overwrite it already do that through the supplied. So why not follow the same pattern?

Reply:
The way that I look at code is that Clock is replacing the `Supplier<Date>`, not the `Date`.

My understanding of the existing code are: 
1. The existing pattern was introduced because there was no way to provide different `Date` suppliers (one giving a skewed Date, another giving an older date,...etc). The solution was a `Supplier<Date>` . 
2. The existing code for some reason chose to put the supplier in the reconcile signature of each reconciler, so ending up with something [like this ](https://github.com/strimzi/strimzi-kafka-operator/blob/main/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L152-L162) and I can't see an obvious reason behind that as opposed to setting the supplier inside the reconciler instance . I'm not sure if there's a benefit of setting the dependency in the resolve signature as opposed to relying on a setter.
   ```java
        reconcileState.initialStatus()
                // Preparation steps => prepare cluster descriptions, handle CA creation or changes
                .compose(state -> state.reconcileCas(this::dateSupplier))
                .compose(state -> state.versionChange())

                // Run reconciliations of the different components
                .compose(state -> featureGates.useKRaftEnabled() ? Future.succeededFuture(state) : state.reconcileZooKeeper(this::dateSupplier))
                .compose(state -> state.reconcileKafka(this::dateSupplier))
                .compose(state -> state.reconcileEntityOperator(this::dateSupplier))
                .compose(state -> state.reconcileCruiseControl(this::dateSupplier))
                .compose(state -> state.reconcileKafkaExporter(this::dateSupplier))
                .compose(state -> state.reconcileJmxTrans())
```


So because point 1. and regardless of point 2, I felt that the only reason for replacing `Supplier<Date>` with `Supplier<Clock>` is to stick to the design out of a necessity to provide tailor a Date supplier, that necessity no longer exists with Clock which is a Date supplier that's provided by the standard lib.

i.e. it felt to me,  the straight forward way if this code would've been written from scratch to provide different Date suppliers is IMHO to compose the reconcilers with a Clock, not a Supplier<Clock>.

Reply:
Well, you do not necessarily need a supplier of course. But you can pass the Clock instead of it. That is what I thought. When you look at the other classes, there are only few methods which use the DateSupplier.

Reply:
@scholzj to make this more concrete, I've refactored the code to show what I'm suggesting. Can you please check and maybe show me a code suggestion just to make sure I understand correctly ?

Reply:
Done